### PR TITLE
#21743: FlattenMerge should propagate outer attributes

### PR DESCRIPTION
**Pull request by drewhk**
- Added Materializer.materialize() version that takes explicit initial attributes

### CODE REVIEW

**Comment by johanandren** [comment]
Add to the flattenMerge docs that this happens?
More stages that materialize inner streams that should get the same treatment?

**Comment by drewhk** [comment]
> Add to the flattenMerge docs that this happens?

True.

> More stages that materialize inner streams that should get the same treatment?

Are there any others?

**Comment by johanandren** [comment]
Maybe it is just `flatMapConcat`, that is the same implementation, didn't know. NVM.

**Comment by drewhk** [comment]
flatMapConcat == flatMapMerge(breadth = 1)

**Review by johanandren — APPROVED** [review]
Awesomesauce!

File: project/MiMa.scala [hunk 1]
```diff
@@ -1016,6 +1016,9 @@ object MiMa extends AutoPlugin {
         ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.actor.VirtualPathContainer.this"),
         ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.remote.RemoteSystemDaemon.this")
 
+      ),
+      "2.4.12" -> Seq(
+        ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.stream.Materializer.materialize")
```
Comment (ktoso) [inline project/MiMa.scala]: Which is OK since we don't uarantee anything for materialized compat - shall we ping Kam about this though?

**Review by ktoso — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**